src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,538 @@
+
+In your newly created content type, use the **Add Field** button to add 5 new fields with the following parameters:
+
+1. Text field
+    - **Name:** title
+    - **API identifier:** `title`
+    (leave the other parameters as their defaults)
+2. Date and time field
+    - **Name:** date
+    - **API identifier:** `date`
+3. Text field
+    - **Name:** slug
+    - **API identifier:** `slug`
+    (leave the other parameters as their defaults)
+4. Text field
+    - **Name:** description
+    - **API identifier:** `description`
+5. Rich text field
+    - **Name:** content
+    - **API identifier:** `content`
+
+Click **Save** to save your changes. In the **Content** section of your Contentful space, create a new entry by clicking the **Add Entry** button. Then, fill in the fields:
+
+- **Title:** `Astro is amazing!`
+- **Slug:** `astro-is-amazing`
+- **Description:** `Astro is a new static site generator that is blazing fast and easy to use.`
+- **Date:** `2022-10-05`
+- **Content:** `This is my first blog post!`
+
+Click **Publish** to save your entry. Feel free to add as many blog posts as you want, then switch to your favorite code editor to start hacking with Astro!
+
+### Displaying blog posts previews
+
+Create a new interface called `BlogPost` and add it to your `contentful.ts` file in `src/lib/`. This interface will match the fields of your blog post content type in Contentful. You will use it to type your blog post entries.
+
+```ts title="src/lib/contentful.ts" ins={2,4-10}
+import contentful from "contentful";
+import type { Document } from "@contentful/rich-text-types";
+
+export interface BlogPost {
+  title: string;
+  date: string;
+  description: string;
+  content: Document;
+  slug: string;
+}
+
+export const contentfulClient = contentful.createClient({
+  space: import.meta.env.CONTENTFUL_SPACE_ID,
+  accessToken: import.meta.env.DEV
+    ? import.meta.env.CONTENTFUL_PREVIEW_TOKEN
+    : import.meta.env.CONTENTFUL_DELIVERY_TOKEN,
+  host: import.meta.env.DEV ? "preview.contentful.com" : "cdn.contentful.com",
+});
+
+```
+
+Next, go to your `index.astro` file in `src/pages/` and import your `BlogPost` interface and  `contentfulClient` from `src/lib/contentful.ts`. 
+
+Using the `getEntries` method from `contentfulClient`, fetch all the entries with a content type of `blogPost`. Pass the `BlogPost` interface to the `getEntries` generic to type your response. 
+
+```astro title="src/pages/index.astro"
+---
+import { contentfulClient } from "../lib/contentful";
+import type { BlogPost } from "../lib/contentful";
+
+const entries = await contentfulClient.getEntries<BlogPost>({
+  content_type: "blogPost",
+});
+---
+```
+
+The items property of `entries` is an array of blog posts. You can use `entries.items.map` to iterate over the array and format the response. In this example, we will destructure the `fields` property of each item and reformat the date to a more readable format.
```

Comment:
```suggestion
This fetch call will return an array of your blog posts at `entries.items`. Use can use `map()` to create a new array (`posts`)  that formats your returned data.

The example below returns the `items.fields` properites from our Content model to create a blog post preview, and at the same time, reformats the date to a more readable format.
```
I don't think the text here has to be quite so jargony. I think it's ok just to describe what is happening at the user level, not the technical term level. Something like the above maybe?